manifest-build-tools/application/update_manifest.py

```diff
@@ -0,0 +1,416 @@
+#!/usr/bin/env python
+
+# Copyright 2015, EMC, Inc.
+
+
+import argparse
+import datetime
+import json
+import os
+import re
+import subprocess
+import sys
+import tempfile
+import shutil
+
+#pylint: disable=relative-import
+import config
+from RepositoryOperator import RepoOperator
+from manifest import Manifest
+
+
+class UpdateManifest(object):
+
+    def __init__(self):
+        """
+        __repo - the repository url being updated in the manifest
+        __branch - the branch name associated with __repo
+        __sliced_branch - the branch name sliced at any forward slashes
+        __commit - The commit id associated with the __repo and __branch that we want to update
+        __manifest_repository_url - the repository url the points to the collection of manifest files
+        __manifest_file - the desired manifest file to update which resides in __manifest_repository_url
+        __cleanup_directories - the path/name of directories created are appended here for cleanup in task_cleanup
+        __git_credentials - url, credentials pair for the access to github repos
+        quiet - used for testing to minimize text written to a terminal
+        repo_operator - Class instance of RepoOperator
+        :return: None
+        """
+        self.__repo = None
+        self.__branch = None
+        self.__sliced_branch = None
+        self.__commit = None
+        self.__manifest_repository_url = None
+        self.__manifest_file = None
+        self.__cleanup_directories = []
+        self.__git_credentials = None
+        self.__updated_manifest = None
+        self.__dryrun = False
+        self.quiet = False
+        self.repo_operator = RepoOperator()
+
+    #pylint: disable=no-self-use
+    def parse_args(self, args):
+        """
+        Take in values from the user.
+        Repo, branch, and manifest_repo are required. This exits if they are not given.
+        :return: Parsed args for assignment
+        """
+        parser = argparse.ArgumentParser()
+        parser.add_argument('-n', '--dryrun',
+                            help="Do not commit any changes, just print what would be done",
+                            action="store_true")
+        parser.add_argument("--repo",
+                            help="Git url to match for updating the commit-id",
+                            action="store")
+        parser.add_argument("--branch",
+                            help="The target branch for the named repo",
+                            action="store")
+        parser.add_argument("--manifest_repo",
+                            help="The manifest repository URL.",
+                            action="store")
+        parser.add_argument("--commit",
+                            help="OPTIONAL: The commit id to target an exact version",
+                            action="store")
+        parser.add_argument("--manifest_file",
```

Comment:
The default value is empty object.
The parameter is the target manifest file to update, for example, rackhd-release-1.2.0
If left it empty, the script will search through all the manifest files to check and update the matched manifest file.